state-chain/amm/src/lib.rs

```diff
@@ -57,6 +57,14 @@ impl<LiquidityProvider: Clone + Ord> PoolState<LiquidityProvider> {
 		})
 	}
 
+	/// Returns the Range Order sub-pool's current price.
+	/// SwapDirection is ignored as the price are the same for both directions.
+	pub fn current_range_order_pool_price(&mut self) -> Option<(Price, SqrtPriceQ64F96, Tick)> {
```

Comment:
So we want to just use the value in range_orders::PoolState::current_sqrt_price (which is not Option), because this is for calculating range order values, not for swaps.

Reply:
Done!